Refactoring Website

wahyuwiyoko opened the pull request:
I'm refactoring some codes without changing much of the layout.

Here is what I have done:

- Use semantic HTML
- Add `description` meta for SEO
- Use `class` instead of `id` for styling
- Delete the `app.js` and move the script to `index.html` file because it's only have a few code
- Remove the `snap-scrolling` style because can overflow the scrollbar
- Self-host the Roboto fonts for a better performance
- Compress some images

Here is the difference from the score:

**Before:**
![firefox_MtnoYCcWsV](https://user-images.githubusercontent.com/109977275/184129946-b1fa2477-226c-411a-8f0f-3d227330e087.png)

**After:**
![firefox_9sueLwrQkf](https://user-images.githubusercontent.com/109977275/184129978-f340f6c4-1d6c-4649-ae95-5d5cd0c35bf8.png)

Here is the demo of my changes: https://wahyuwiyoko.github.io/libre-tube.github.io/

Better to test it on desktop and mobile before merge this PR.
I hope I don't break up the site layout :grin:

Commit:
Code refactor, compress images, self-host Roboto fonts

Commit:
Add alt for images

Commit:
Change h3 to h2, minor font changes

ghost commented:
Oh and one thing, I changed the `assets` directory name to `images`. So I saw there's [LibreTube README](https://github.com/libre-tube/LibreTube/blob/master/README.md?plain=1) are using that URL. I will make PR to fix the README if this merged.

Commit:
Add wrap to download links, 90% width of features

Bnyro commented:
Hey, thank you for your interest in contributing and your work concerning the website!

I just tested your demo and would suggest some minor changes:

![image](https://user-images.githubusercontent.com/82752168/184180656-c37e4d56-c25d-4e85-85e2-52b5bb511984.png)

1. Previously, the "Privacy Simplified" `h1` has been center aligned while the social links and donation button have been aligned to the bottom. I would suggest to keep it like that, it feels pretty much squeezed when being centered completely.

![image](https://user-images.githubusercontent.com/82752168/184181725-1b3eeacd-9b75-40f2-aefb-4404b28cfe3f.png)

![image](https://user-images.githubusercontent.com/82752168/184181474-930e54e5-3ea0-462d-b2fa-d32c3e1bb41b.png)

2. I'm still a fan of the top bar padding we had here, I would suggest to increase the header padding again to make it feel less tight.

![image](https://user-images.githubusercontent.com/82752168/184182377-20908e77-961f-4244-b416-3278af4135c4.png)

3. The width of the articles in the FAQ section could be decreased/their margin could be increased in my opinion, it just looks really stretched on larger screens now since the questions are really short.

Regards and thanks again for your work!

Bnyro commented:
And one last thing I forgot to mention:
What exactly was the issue with snap-scrolling?
It got added by another PR and I'm pretty happy with it, I don't think we need to remove it :thinking:

ghost commented:
> 1. Previously, the "Privacy Simplified" `h1` has been center aligned while 
> the social links and donation button have been aligned to the bottom. 
> I would suggest to keep it like that, it feels pretty much squeezed when being centered completely.
>  2. I would suggest to increase the header padding again to make it feel less tight.

I'm increased the `padding-top` of the `#about` style (the container) to `6rem` so it's goes a bit to bottom. And the `header` padding to `2rem` (top and bottom) and `5rem` (left and right). Here is how it's look like:
![firefox_mz7my5WaZj](https://user-images.githubusercontent.com/109977275/184238629-5c77cb1b-9fbc-4338-bc59-d9a2360f440c.png)

>  3. The width of the articles in the FAQ section could be decreased/their margin 
>  could be increased in my opinion, it just looks really stretched on larger screens now since the questions are really short.

I changed the width of `.faq` to 60%. Here is how it's look like:
![firefox_OZmDKnLW1Q](https://user-images.githubusercontent.com/109977275/184200428-6f59c909-304a-481d-92d2-763f7bc3fe1f.png)

~~What exactly was the issue with snap-scrolling?~~

~~I had problem with that because the `main` and `footer` are separated that caused the scrollbar overflow. Only the `main` had `snap-scrolling` class. It will make 2 vertical scrollbar and the `footer` behavior looks like `fixed` position if I scrolled.~~
 
I have added the snap-scrolling back now 😄 
Check the new changes on the demo: https://wahyuwiyoko.github.io/libre-tube.github.io/

wahyuwiyoko commented on style.css:
```diff
@@ -13,151 +43,147 @@ html {
   display: flex;
   justify-content: space-between;
   align-items: center;
+  padding: 1.5rem 2.5rem 1.5rem 2rem;
```
Tried change to `padding: 2rem 5rem;`

wahyuwiyoko commented on style.css:
```diff
@@ -13,151 +43,147 @@ html {
-  margin-top: 42vh;
+/* About Section */
+#about {
+  padding-top: 1rem;
```
The `Privacy Simplified` container goes a bit to bottom with `padding-top: 4rem;`

wahyuwiyoko commented on style.css:
```diff
@@ -13,151 +43,147 @@ html {
-  margin: 0.7em 0;
+/* FAQ Section */
+.faq {
+  width: 90%;
```
Decreased the `.faq` container width to 60%

Commit:
Change about section padding to 6rem

Commit:
Minor changes

Add snap-scrolling, change header padding, change about section height, change faq width

Commit:
Add snap-scrolling, change header padding, about section height, faq width

Bnyro commented:
Tested, works well now, thank you for your help!
